src/wazuh_modules/agent_upgrade/manager/wm_agent_upgrade_tasks_callbacks.c

```diff
@@ -0,0 +1,121 @@
+
+    if (task_module_response && (task_module_response->type == cJSON_Array) && (agents == cJSON_GetArraySize(task_module_response))) {
+        // Parse task module responses
+        while( cJSON_GetArraySize(task_module_response) && (error == OS_SUCCESS) ) {
+            cJSON *task_response = cJSON_DetachItemFromArray(task_module_response, 0);
+            if (success_callback) {
+                // A callback has been defined, process it with the callback
+                cJSON *callback_object = success_callback(&error, task_response);
+                if (callback_object) {
+                    cJSON_AddItemToArray(json_response, callback_object);
+                }
+                if (callback_object != task_response) {
+                    cJSON_Delete(task_response);
+                }
+            } else {
+                cJSON_AddItemToArray(json_response, task_response);
+            }
+            
+        }
+    } else {
+        error = OS_INVALID;
+    }
+
+    if (error) {
+        for(int i = 0; i < agents; i++) {
+            cJSON *agent_json = cJSON_GetObjectItem(cJSON_GetArrayItem(task_module_request, i), "agent");
+
+            if (agent_json && (agent_json->type == cJSON_Number)) {
+                int agent_id = agent_json->valueint;
+                if (error_callback) {
+                    error_callback(agent_id);
+                } 
+                error_json = wm_agent_upgrade_parse_response_message(WM_UPGRADE_TASK_MANAGER_COMMUNICATION, upgrade_error_codes[WM_UPGRADE_TASK_MANAGER_COMMUNICATION], &agent_id, NULL, NULL);
+                cJSON_AddItemToArray(json_response, error_json);
+            }
+        }
+        mterror(WM_AGENT_UPGRADE_LOGTAG, WM_UPGRADE_INVALID_TASK_MAN_JSON);
+    }
+
+    cJSON_Delete(task_module_response);
+
+    return error;
+}
+
+cJSON* wm_agent_upgrade_upgrade_success_callback(int *error, cJSON* input_json) {
+    int agent_id;
+    int task_id;
+    char *data = NULL;
+    cJSON *response = NULL;
+    
+    if (wm_agent_upgrade_validate_task_ids_message(input_json, &agent_id, &task_id, &data)) {
+        if(task_id) {
+            // Store task_id
+            wm_agent_upgrade_insert_task_id(agent_id, task_id);
+            response = input_json;
+        } else {
+            // Remove from table since upgrade will not be started
+            wm_agent_upgrade_remove_entry(agent_id);
+            response = wm_agent_upgrade_parse_response_message(WM_UPGRADE_TASK_MANAGER_FAILURE, data, &agent_id, NULL, NULL);
+        }
+    } else {
+        // We cannot know which agent is the one failing so we have to abort the whole process
+        *error = OS_INVALID;
+    }
+    os_free(data);
+    return response;
+}
+
+cJSON* wm_agent_upgrade_update_status_success_callback(int *error, cJSON* input_json) {
+    cJSON *agent_json = cJSON_GetObjectItem(input_json, task_manager_json_keys[WM_TASK_AGENT_ID]);
```

Comment:
Get and validate the `agent_id` in `wm_agent_upgrade_validate_task_status_message`

Reply:
Ok